distribution/tools/entitlement-runtime/src/main/java/org/elasticsearch/entitlement/runtime/api/FlagEntitlement.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.api;
+
+/**
+ * A simple kind of entitlement that is either present or absent, with no additional structure.
+ */
+public enum FlagEntitlement implements Entitlement {
```

Comment:
I think the actual entitlement definitions will need some work...is this an interim implementation?

Reply:
How do we model different types of entitlement that can't be represented just by an enum (eg file permissions)?

Reply:
Actually @jdconrad is working on that, I think he has ideas and probably some POJOs to express entitlements/policies ready already (or soon)

Reply:
They'd be additional subtypes of `Entitlement`. In the POC I had `FileEntitlement` for example, with additional fields.